### ISSUE TITLE
null ContainerInfo while processing matches

### ISSUE THREAD

**Issue by smcquay** [jenkinsci/mesos-plugin#105]
I was following the instructions in the README.md and I was unable to get tasks to start. My setup:

All on same machine:

- mesos-master
- mesos-slave
- maven hpi:run

Then after I configure a task to run on a node with "mesos" label, I get a NullPointerException in JenkinsScheduler.java. Regardless the local jenkins instance runs, and I see this offer back and forth in the master logs (immediate offer and reclamation):

```log
I0515 18:08:20.104230 21788 master.cpp:3843] Sending 1 offers to framework 20150515-164349-1138285329-5050-21768-0008 (Jenkins Scheduler) at scheduler-90d41
353-bd73-44d1-aac7-de848389c38b@127.0.1.1:52112
I0515 18:08:20.106294 21786 master.cpp:2344] Processing reply for offers: [ 20150515-164349-1138285329-5050-21768-O124 ] on slave 20150515-131535-1138285329
-5050-7398-S0 at slave(1)@127.0.1.1:5051 (simcloud-spare-node00) for framework 20150515-164349-1138285329-5050-21768-0008 (Jenkins Scheduler) at scheduler-9
0d41353-bd73-44d1-aac7-de848389c38b@127.0.1.1:52112
I0515 18:08:20.106849 21785 hierarchical_allocator_process.hpp:563] Recovered cpus(*):24; mem(*):71332; disk(*):472; ports(*):[31000-32000] (total allocatab
le: cpus(*):24; mem(*):71332; disk(*):472; ports(*):[31000-32000]) on slave 20150515-131535-1138285329-5050-7398-S0 from framework 20150515-164349-113828532
9-5050-21768-0008

```

If I modify the code to look as such:

```diff
diff --git a/src/main/java/org/jenkinsci/plugins/mesos/JenkinsScheduler.java b/src/main/java/org/jenkinsci/plugins/mesos/JenkinsScheduler.java
index 610ae98..edae2ab 100644
--- a/src/main/java/org/jenkinsci/plugins/mesos/JenkinsScheduler.java
+++ b/src/main/java/org/jenkinsci/plugins/mesos/JenkinsScheduler.java
@@ -303,7 +303,14 @@ public class JenkinsScheduler implements Scheduler {
     if (cpus < 0) LOGGER.fine("No cpus resource present");
     if (mem < 0)  LOGGER.fine("No mem resource present");
 
-    boolean hasPortMappings = request.request.slaveInfo.getContainerInfo().hasPortMappings();
+    Mesos.SlaveRequest sr = request.request;
+    MesosSlaveInfo si = sr.slaveInfo;
+    MesosSlaveInfo.ContainerInfo ci = si.getContainerInfo();
+    if (ci == null) {
+      LOGGER.severe("ContainerInfo is null?!");
+      return true;
+    }
+    boolean hasPortMappings = ci.hasPortMappings();
     boolean hasPortResources = ports != null && !ports.isEmpty();
 
     if (hasPortMappings && !hasPortResources) {
```

the job runs. However it seems like the wrong approach. I'm very fresh to mesos; is there something obvious I'm doing that would cause the plugin not to get a complete ContainerInfo?

**Comment by arjang** [comment]
Have you solved this problem? i am getting the same error

**Comment by smcquay** [comment]
I have not solved it aside from what seems to be the wrong approach outlined above.

**Comment by arjang** [comment]
if you tick 'Use Docker Containerizer' and fill the docker fields, you dont get null pointer exception anymore. i assume this plugin sets this option to true by default.

**Comment by maselvaraj** [comment]
thanks @smcquay. I've fixed this is the master branch. Will be released in 0.8.0.
